drivers/encx24j600/encx24j600.c

```diff
@@ -0,0 +1,424 @@
+
+/* dev_eth interface */
+static void _isr(dev_eth_t *dev);
+static int _init(dev_eth_t *dev);
+static int _send(dev_eth_t *dev, char *pkt, int len);
+static void _get_mac_addr(dev_eth_t *dev, uint8_t* buf);
+static int _recv(dev_eth_t *encdev, char* buf, int len);
+
+static inline int _get_promiscous(dev_eth_t *dev) {
+    return 1;
+}
+
+static inline int _set_promiscous(dev_eth_t *dev, int value) {
+    return 1;
+}
+
+const static eth_driver_t eth_driver_encx24j600 = {
+    .init = _init,
+    .send = _send,
+    .recv = _recv,
+    .get_mac_addr = _get_mac_addr,
+    .get_promiscous = _get_promiscous,
+    .set_promiscous = _set_promiscous,
+    .isr = _isr,
+};
+
+static inline void lock(encx24j600_t *dev) {
+    mutex_lock(&dev->mutex);
+}
+
+static inline void unlock(encx24j600_t *dev) {
+    mutex_unlock(&dev->mutex);
+}
+
+void encx24j600_setup(encx24j600_t *dev, spi_t spi, gpio_t cs, gpio_t int_pin)
+{
+    dev->ethdev.driver = &eth_driver_encx24j600;
+    dev->spi = spi;
+    dev->cs = cs;
+    dev->int_pin = int_pin;
+    dev->rx_next_ptr = RX_BUFFER_START;
+
+    mutex_init(&dev->mutex);
+}
+
+static void encx24j600_isr(void *arg)
+{
+    encx24j600_t *dev = (encx24j600_t *) arg;
+
+    /* disable interrupt line */
+    gpio_irq_disable(dev->int_pin);
+
+    /* call dev_eth hook */
+    dev_eth_isr((dev_eth_t*) dev);
+}
+
+static void _isr(dev_eth_t *encdev)
+{
+    encx24j600_t *dev = (encx24j600_t *) encdev;
+
+    uint16_t eir;
+
+    lock(dev);
+    cmd(dev, CLREIE);
+
+    eir = reg_get(dev, EIR);
+
+    if (eir & PKTIF) {
+        receive_packets(dev);
+    }
+
+    if (eir & LINKIF) {
+        uint16_t estat = reg_get(dev, ESTAT);
+        dev_eth_linkstate_handler(encdev, estat & PHYLNK);
+    }
+
+    /* drop all flags */
+    reg_clear_bits(dev, EIR, LINKIF);
+
+    /* re-enable interrupt */
+    gpio_irq_enable(dev->int_pin);
+    cmd(dev, SETEIE);
+
+    unlock(dev);
+}
+
+static inline void enc_spi_transfer(encx24j600_t *dev, char *out, char *in, int len)
+{
+    gpio_clear(dev->cs);
+    spi_transfer_bytes(dev->spi, out, in, len);
+    gpio_set(dev->cs);
+}
+
+static inline uint16_t reg_get(encx24j600_t *dev, uint8_t reg)
+{
+    char cmd[4] = { RCRU, reg, 0, 0 };
+    char result[4];
+
+    enc_spi_transfer(dev, cmd, result, 4);
+
+//    DEBUG("encx24j600_reg_get: got 0x%x from register 0x%x\n", (result[2] | (result[3] << 8)), reg);
+
+    return result[2] | (result[3] << 8);
+}
+
+static void phy_reg_set(encx24j600_t *dev, uint8_t reg, uint16_t value) {
+    reg_set(dev, MIREGADR, reg | (1<<8));
+    reg_set(dev, MIWR, value);
+}
+
+static void cmd(encx24j600_t *dev, char cmd) {
+    gpio_clear(dev->cs);
+    spi_transfer_byte(dev->spi, cmd, NULL);
+    gpio_set(dev->cs);
+}
+
+static void cmdn(encx24j600_t *dev, uint8_t cmd, char *out, char *in, int len) {
+    gpio_clear(dev->cs);
+    spi_transfer_byte(dev->spi, cmd, NULL);
+    spi_transfer_bytes(dev->spi, out, in, len);
```

Comment:
Please replase it (L168, L169) with spi_transfer_regs(dev->spi, cmd, out, in, len);